File: Easy/Power-of-thor-episode-1/Power-of-thor-episode-1.vb
Module Player
    Sub Main ()
        
        Dim inputs as String()
        Dim lightX as Integer ' the X position of the light of power
        Dim lightY as Integer ' the Y position of the light of power
        Dim initialTX as Integer ' Thor's starting X position
        Dim initialTY as Integer ' Thor's starting Y position
        inputs = Console.ReadLine().Split(" ")
        lightX = inputs(0)
        lightY = inputs(1)
        initialTX = inputs(2)
        initialTY = inputs(3)

        ' game loop
        While True
            Dim remainingTurns as Integer
            remainingTurns = Console.ReadLine()

            If lightY > initialTY Then
                initialTY += 1
                Console.Write("S")
            ElseIf lightY < initialTY Then
                initialTY -= 1
                Console.Write("N")
            End If
            If lightX > initialTX Then
                initialTX += 1
                Console.Write("E")
            ElseIf lightX < initialTX Then
                initialTX -= 1
                Console.Write("W")
            End If
            Console.WriteLine()
        End While
    End Sub
End Module

File: Easy/Power-of-thor-episode-1/Power-of-thor-episode-1.swift
let inputs = (readLine()!).split(separator: " ").map(String.init)
let lightX = Int(inputs[0])! // the X position of the light of power
let lightY = Int(inputs[1])! // the Y position of the light of power
var TX = Int(inputs[2])! // Thor's starting X position
var TY = Int(inputs[3])! // Thor's starting Y position

while true {
    let _ = Int(readLine()!)!
    var str : String = ""
    if lightY > TY {
        TY += 1; str += "S"
    } else if lightY < TY {
        TY -= 1; str += "N"
    }
    if lightX > TX {
        TX += 1; str += "E"
    } else if lightX < TX {
        TX -= 1; str += "W"
    }
    print(str)
}
